src/google/protobuf/descriptor.h

```diff
@@ -615,7 +615,7 @@ class PROTOBUF_EXPORT Descriptor : private internal::SymbolBase {
   GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(Descriptor);
 };
 
-PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(Descriptor, 136);
+PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(Descriptor, 136)
```

Comment:
The macro is defined empty:
`#if !defined(PROTOBUF_INTERNAL_CHECK_CLASS_SIZE)
#define PROTOBUF_INTERNAL_CHECK_CLASS_SIZE(t, expected)
#endif`

which means the double ; comes from the fact that we have a regular statement before the macro, then emptyness (the macro) and then another ; after that.

Reply:
you could change the empty definition to 

do {} while (false)

Reply:
since the macro's are used outside of a function scope, a regular function calling or piece of code won't compile there ("expected constructor, destructor, or type conversion before ';' token"). Putting an empty class or struct definition is also not possible since the macro is used multiple times so then you get redefinition errors. 
The macro seems like something used internally by google, is that correct?

Reply:
you can change them to `static_assert(true, "")` which will be valid at all scopes.